### PR TITLE
feat(iot-actions): add the action to put CloudWatch Logs

### PR DESCRIPTION
**Pull request by yamatatsu**
I'm trying to implement aws-iot L2 Constructs.

This PR is one of steps after following PR: 
- https://github.com/aws/aws-cdk/pull/16681#issuecomment-942233029

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by yamatatsu** [comment]
@skinny85 
Thank you for your reply! You've made me feel more confident 😀

**Comment by yamatatsu** [comment]
@madeline-k 
This is ready to be reviewed now!

File: packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-logs-action.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import * as logs from '@aws-cdk/aws-logs';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for CloudWatch Logs.
+ */
+export interface CloudwatchLogsActionProps {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-logs-action.ts:9]: Same here:

```suggestion
export interface CloudWatchLogsActionProps {
```

File: packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-logs-action.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  constructor(
+    private readonly logGroup: logs.ILogGroup,
+    props: CloudwatchLogsActionProps = {},
+  ) {
+    this.role = props.role;
+  }
+
+  bind(rule: iot.ITopicRule): iot.ActionConfig {
+    const role = this.role ?? singletonActionRole(rule);
+    this.logGroup.grant(role, 'logs:CreateLogStream', 'logs:DescribeLogStreams', 'logs:PutLogEvents');
```
Comment (skinny85) [inline packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-logs-action.ts:37]: `'logs:CreateLogStream'` and `'logs:PutLogEvents'` are granted by `logGroup.grantWrite(role);`. Let's maybe use that for those 2?

File: packages/@aws-cdk/aws-iot-actions/test/cloudwatch-logs/cloudwatch-logs-action.test.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+test('can set role', () => {
+  // GIVEN
+  const stack = new cdk.Stack();
+  const topicRule = new iot.TopicRule(stack, 'MyTopicRule', {
+    sql: iot.IotSql.fromStringAsVer20160323("SELECT topic(2) as device_id FROM 'device/+/data'"),
+  });
+  const logGroup = new logs.LogGroup(stack, 'MyLogGroup');
+  const role = new iam.Role(stack, 'MyRole', {
+    assumedBy: new iam.ServicePrincipal('iot.amazonaws.com'),
+  });
```
Comment (skinny85) [inline packages/@aws-cdk/aws-iot-actions/test/cloudwatch-logs/cloudwatch-logs-action.test.ts:87]: If you use `iam.Role.fromRoleArn()` instead, the assertions will be a little simpler to formulate I think.

**Review by skinny85 — DISMISSED** [review]
Looks great, as always @yamatatsu! A few tiny comments before we merge this in 🙂.

File: packages/@aws-cdk/aws-iot-actions/test/cloudwatch-logs/cloudwatch-logs-action.test.ts (hunk 3 above)
Comment (yamatatsu) [reply packages/@aws-cdk/aws-iot-actions/test/cloudwatch-logs/cloudwatch-logs-action.test.ts:87]: Cool!😲

File: packages/@aws-cdk/aws-iot-actions/README.md [hunk 4]
```diff
@@ -48,3 +48,21 @@ new iot.TopicRule(this, 'TopicRule', {
...
 ```
+
+## Put logs to CloudWatch Logs
+
+The code snippet below creates an AWS IoT Rule that put logs to CloudWatch Logs
+when it is triggered.
+
+```ts
+import * as iot from '@aws-cdk/aws-iot';
+import * as actions from '../../lib';
```
Comment (skinny85) [inline packages/@aws-cdk/aws-iot-actions/README.md:59]:
```suggestion
import * as actions from '@aws-cdk/aws-iot-actions';
```

**Review by skinny85 — APPROVED** [review]
One last change needed. I'll do it myself though 😉.

**Review by skinny85 — APPROVED** [review]
Awesome stuff as always @yamatatsu!

**Pull request merged**